code/obj/item/clothing/hats.dm

```diff
@@ -1961,6 +1961,121 @@ TYPEINFO(/obj/item/clothing/head/lesbian_hat)
 	item_state = "space_replica"
 	desc = "A replica of an old space helmet. Looks spaceworthy regardless."
 
+/obj/item/clothing/head/giraffehat
+	name = "giraffe hat"
+	desc = "Great for finally reaching those tender tree-top leaves."
+	icon = 'icons/obj/large/32x48.dmi'
+	icon_state = "giraffehat"
+	item_state = "giraffehat"
+
+/obj/item/clothing/head/rhinobeetle
+	name = "rhino beetle helm"
+	desc = "A lightweight helm styled like a rhinocerous beetle's horn. Not sturdy enough for you to flip your enemies over with, sadly."
+	icon_state = "rhinobeetle"
+	item_state = "rhinobeetle"
+
+/obj/item/clothing/head/stagbeetle
+	name = "stag beetle helm"
+	desc = "A lightweight helm styled like a stag beetle's mandibles. Not actually a functional set of grabbers, unfortunately."
+	icon_state = "stagbeetle"
+	item_state = "stagbeetle"
+
+TYPEINFO(/obj/item/clothing/head/elephanthat)
+	random_subtypes = list(/obj/item/clothing/head/elephanthat/gold,
+		/obj/item/clothing/head/elephanthat/green,
+		/obj/item/clothing/head/elephanthat/pink,
+		/obj/item/clothing/head/elephanthat/blue
+	)
+ABSTRACT_TYPE(/obj/item/clothing/head/elephanthat)
+/obj/item/clothing/head/elephanthat
+	name = "elephant hat"
+	desc = "Quite cozy, if you don't mind the trunk smacking you in the face when you walk."
+	icon_state = "elephant-gold"
+	item_state = "elephant-gold"
+
+	gold
+		icon_state = "elephant-gold"
+		item_state = "elephant-gold"
+
+	blue
+		icon_state = "elephant-blue"
+		item_state = "elephant-blue"
+
+	pink
+		icon_state = "elephant-pink"
+		item_state = "elephant-pink"
+
+	green
+		icon_state = "elephant-green"
+		item_state = "elephant-green"
+
+	random
+		New()
+			var/old_loc = src.loc
+       	qdel(src)
+		src = get_random_subtype(/obj/item/clothing/head/elephanthat,TRUE)
+		src.set_loc(old_loc)
+        ..()
```

Comment:
Code here is looking a bit odd, this overriding the qdel proc for this hat and all subtypes likely due to an indentation mistake

In general, do not override src. It's very rare that it is ever necessary, and even in those instances it's pretty much always a bad hack to avoid having to solve the real issue at hand. What you're trying to do though is qdel src and then replace it with the new instance of one of the other subtypes. This is unsafe to do though, as in various instances the creator of one of these random hats may also care about its reference. For example a job trying to equip starter gear would attempt to equip a since deleted hat object to the player

```suggestion
	random
		New()
			. = ..()
			var/obj/item/rand_type = get_random_subtype(/obj/item/clothing/head/elephanthat)
			icon_state = initial(rand_type.icon_state)
			item_state = initial(rand_type.item_state)
```

Reply:
Hmm thats my bad

Reply:
AH I see I see... thank you for clarifying I had a feeling there was a more direct way to do this

Reply:
> Hmm thats my bad. Does this update the description of the random item to be that of the subtype that's picked?

As these already share a common description there was no need.